ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -851,7 +851,9 @@ private void writeMessagesToLocalStoreAndAdvanceTokens(List<ExchangeMetadataResp
               getResponse.getPartitionResponseInfoList().get(partitionResponseInfoIndex);
           responseHandler.onEvent(remoteReplicaInfo.getReplicaId(), partitionResponseInfo.getErrorCode());
           partitionResponseInfoIndex++;
-          if (partitionResponseInfo.getPartition().compareTo(remoteReplicaInfo.getReplicaId().getPartitionId()) != 0) {
+          if (!partitionResponseInfo.getPartition()
```

Comment:
Thats why I have this change. In case of a cluster map where local partition comes from static cluster map(Partition.java), while the remote replicas comes from helix cluster map(AmbryPartition.java), it will not to possible to compare them because they will have different types. However, since replication can happen, the remote replicas will have AmbryPartition as partition type. Hence in order to make sure they have the same partition, matching of just the id should suffice.